addons/account_peppol/models/res_company.py

```diff
@@ -32,6 +56,59 @@ class ResCompany(models.Model):
         compute='_compute_peppol_purchase_journal_id', store=True, readonly=False,
     )
 
+    # -------------------------------------------------------------------------
+    # HELPER METHODS
+    # -------------------------------------------------------------------------
+
+    def _sanitize_peppol_phone_number(self, phone_number):
+        error_message = _(
+            "Please enter the phone number in the correct international format.\n"
+            "For example: +32123456789, where +32 is the country code.\n"
+            "Currently, only European countries are supported.")
+
+        if not phonenumbers:
+            raise ValidationError(error_message)
+
+        if not phone_number.startswith('+'):
+            phone_number = f'+{phone_number}'
+
+        try:
+            phone_nbr = phonenumbers.parse(phone_number)
+        except phonenumbers.phonenumberutil.NumberParseException:
+            raise ValidationError(error_message)
+
+        country_code = phonenumbers.phonenumberutil.region_code_for_number(phone_nbr)
+        if country_code not in ALLOWED_COUNTRIES or not phonenumbers.is_valid_number(phone_nbr):
+            raise ValidationError(error_message)
+
+        phone_fmt = phonenumbers.PhoneNumberFormat.E164
+        return phonenumbers.format_number(phone_nbr, phone_fmt)
+
+    # -------------------------------------------------------------------------
+    # CONSTRAINTS
+    # -------------------------------------------------------------------------
+
+    @api.constrains('account_peppol_phone_number')
+    def _check_account_peppol_phone_number(self):
+        for company in self:
+            if company.account_peppol_phone_number:
+                company._sanitize_peppol_phone_number(company.account_peppol_phone_number)
+
+    @api.constrains('peppol_endpoint')
+    def _check_peppol_endpoint(self):
+        for company in self:
+            if not company.peppol_endpoint or company.peppol_eas not in PEPPOL_ENDPOINT_RULES:
+                continue
+            if company.peppol_eas == '0208' and 'BE' in company.peppol_endpoint:
```

Comment:
I really don't like the explicit stuff for Belgium... If we do that for all countries, it will be a mess with specific rules everywhere :(

Reply:
yes, but the stdnum module accepts numbers prefixed with 'BE' :/ So not sure how to circumvent this in another way. And the peppol participant migration depends on the Belgian endpoint being valid :(